### ISSUE TITLE
The predicted result after conversion is incorrect

### ISSUE THREAD

**Issue by kaidi-jin** [qxcv/caffe2keras#10]
hi, I  meet some problem here.
 I used this tool and converted caffemodel to keras h5 file successfully.But when use the  h5 model to predict  image ,the result is wrong. I can't find the reason. the caffemodel  is mnist(lenet 5) and I have tested it is right. I use theano as backend and image format is channel first .
can any body give me some tips? what should I do?  Thank you very much!!!
 here is the prototxt information:

name: "LeNet"
layer {
  name: "data"
  type: "Input"
  top: "data"
  input_param { shape: { dim: 64 dim: 1 dim: 28 dim: 28 } }
}
layer {
  name: "conv1"
  type: "Convolution"
  bottom: "data"
  top: "conv1"
  param {
    lr_mult: 1
  }
  param {
    lr_mult: 2
  }
  convolution_param {
    num_output: 20
    kernel_size: 5
    stride: 1
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "pool1"
  type: "Pooling"
  bottom: "conv1"
  top: "pool1"
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
  }
}
layer {
  name: "conv2"
  type: "Convolution"
  bottom: "pool1"
  top: "conv2"
  param {
    lr_mult: 1
  }
  param {
    lr_mult: 2
  }
  convolution_param {
    num_output: 50
    kernel_size: 5
    stride: 1
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "pool2"
  type: "Pooling"
  bottom: "conv2"
  top: "pool2"
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
  }
}
layer {
  name: "ip1"
  type: "InnerProduct"
  bottom: "pool2"
  top: "ip1"
  param {
    lr_mult: 1
  }
  param {
    lr_mult: 2
  }
  inner_product_param {
    num_output: 500
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "relu1"
  type: "ReLU"
  bottom: "ip1"
  top: "ip1"
}
layer {
  name: "ip2"
  type: "InnerProduct"
  bottom: "ip1"
  top: "ip2"
  param {
    lr_mult: 1
  }
  param {
    lr_mult: 2
  }
  inner_product_param {
    num_output: 10
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "prob"
  type: "Softmax"
  bottom: "ip2"
  top: "prob"
}

**Comment by qxcv** [comment]
I merged some fixes a week ago that made the code work with TF rather than Theano, but forgot to put in backend checks (it should have warned you if you were using Theano). Reverting back to 27429dd94554281905fd2f331700b4514ae2f71e will probably make it work again.

**Comment by kaidi-jin** [comment]
Thank you for your reply! I will try it. If there are some new result I will show you as soon as possible.
THANK YOU！

**Comment by kaidi-jin** [comment]
hi，yeah, As you suggested, I used your  a week ago's code. This time I use Tensorflow as backend and image format is channel last.  and it works! the result is right! :) 
however,I meet some thing new today.

when use your latest code. the keras out model  have some problem. I still use the mnist model.When convert pooling layer. the out shape is not right. 

the  wrong created model as follow:



Layer (type)                 Output Shape              Param #   
=================================================================
data (InputLayer)            (None, 1L, 28L, 28L)      0         
_________________________________________________________________
conv1 (Conv2D)               (None, 20, 24L, 24L)      520       
_________________________________________________________________
pool1maxpooling (MaxPooling2 (None, 20, 23L, 23L)      0         
_________________________________________________________________
conv2 (Conv2D)               (None, 50, 19L, 19L)      25050     
_________________________________________________________________
pool2maxpooling (MaxPooling2 (None, 50, 18L, 18L)      0         
.........


The right model :
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
data (InputLayer)            (None, 1, 28, 28)         0         
_________________________________________________________________
conv1 (Conv2D)               (None, 20, 24, 24)        520       
_________________________________________________________________
pool1maxpooling (MaxPooling2 (None, 20, 24, 24)        0         
_________________________________________________________________
pool1 (Lambda)               (None, 20, 12, 12)        0         
_________________________________________________________________
......


Hope you can understand what i say.forgive me bad English.
Anyway,thank you for your work. do me a big favor!

**Comment by qxcv** [comment]
That's odd, and I'm not sure what could be causing it. Perhaps it's an inappropriate padding mode? One possible hack: you could try adding an explicit zero-padding layer to your `.prototxt` (before conv1) to make the input large enough that the convolution doesn't reduce its size.

**Comment by kaidi-jin** [comment]
ok~ thank you
